homeassistant/components/switchbee/config_flow.py

```diff
@@ -0,0 +1,133 @@
+    """Validate the user input allows us to connect."""
+
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = CentralUnitAPI(
+        data[CONF_HOST], data[CONF_USERNAME], data[CONF_PASSWORD], websession
+    )
+    try:
+        await api.connect()
+    except SwitchBeeError as exp:
+        _LOGGER.error(exp)
+        if "LOGIN_FAILED" in str(exp):
+            raise InvalidAuth from SwitchBeeError
+
+        raise CannotConnect from SwitchBeeError
+
+    return format_mac(api.mac)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for SwitchBee Smart Home."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Show the setup form to the user."""
+        errors: dict[str, str] = {}
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            )
+
+        try:
+            mac_formated = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+
+        else:
+            await self.async_set_unique_id(mac_formated)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=user_input[CONF_HOST], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for AEMET."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None) -> FlowResult:
+        """Handle options flow."""
+
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        all_devices = [
+            DeviceType.Switch,
+            DeviceType.TimedSwitch,
+            DeviceType.GroupSwitch,
+            DeviceType.TimedPowerSwitch,
+        ]
+
+        data_schema = {
+            vol.Required(
+                CONF_DEVICES,
+                default=self.config_entry.options.get(
+                    CONF_DEVICES,
+                    CONF_DEFUALT_ALLOWED,
+                ),
+            ): cv.multi_select([device.display for device in all_devices]),
+        }
```

Comment:
Device selection / inclusion/exclusion is not a task of an integration. It should provide whatever it can. Users can enable/disable what they like via the Home Assistant interface already. Please remove this.